lib/config_files/base_model.rb

```diff
@@ -38,5 +69,77 @@ def merge_changes
       # TODO: recursive merge
       data.merge(new_data)
     end
+
+    def modify(key, value)
+      # if already set, just change value
+      return false unless data[key]
+
+      data[key] = value
+      true
+    end
+
+    def uncomment(key, value)
+      # Try to find if it is commented out, so we can replace line
+      matcher = Matcher.new(
+        collection:    "#comment",
+        value_matcher: /#{key}\s*=/
```

Comment:
Is this generic enough? Looks like it will only work with file formats using `=` (like sysconfig or ini files).

I'm also not sure about situations in which similar keys exist, like having a key called `one` and other called `another_one`. When searching for the former, will not both keys match the expression?